**Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170.  
Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175.   
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.**  

In [1]:
import math
import numpy as np
from numpy import linalg as LA
from scipy import stats 
import statistics

Расчитываем интервальную оценку для разности средних арифметических по формуле: 
 
$\bar{x}_1 - \bar{x}_2 \pm t{\frac{\alpha}{2}} \sqrt{\frac{D}{n_1} + \frac{D}{n_2}}$

где:  
$\bar{x}_1$ - среднее арифметическое в группе 1  
$\bar{x}_2$ - среднее арифметическое в группе 2  
$D$ - средняя дисперсия обеих групп  
$n_1$ - число элементов в группе 1  
$n_2$ - число элементов в группе 2  

При этом средняя дисперсия обеих групп расчитывается по формуле:  
$D=\frac{1}{2}(D_1+D_2)$  

где:  
$D_1$ - дисперсия группы 1  
$D_2$ - дисперсия группы 2

Вводим данные

In [2]:
mothers = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
daughters = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])

Расчитываем разность средних арифметических $\bar{x}_1 - \bar{x}_2$:

In [3]:
difference_of_means = np.mean(mothers) - np.mean(daughters)
print(f'Среднее арифметическое роста матерей - {np.mean(mothers): .2f} см.')
print(f'Среднее арифметическое роста дочерей - {np.mean(daughters): .2f} см.')
print(f'Разность средних арифметических составляет - {difference_of_means: .2f} см.')

Среднее арифметическое роста матерей -  168.10 см.
Среднее арифметическое роста дочерей -  166.20 см.
Разность средних арифметических составляет -  1.90 см.


Находим несмещенные дисперсии для каждой выборки и среднюю дисперсию обеих групп:

In [4]:
D1=np.var(mothers, ddof=1)
D2=np.var(daughters, ddof=1)
print(f'Дисперсия выборки роста матерей - {D1: .4f}')
print(f'Дисперсия выборки роста дочерей - {D2: .4f}')
D=0.5*(D1+D2)
print(f'Средняя дисперсия обеих групп - {D: .4f}')

Дисперсия выборки роста матерей -  60.1000
Дисперсия выборки роста дочерей -  91.0667
Средняя дисперсия обеих групп -  75.5833


Расчитываем $\sqrt{\frac{D}{n_1} + \frac{D}{n_2}}$:

In [5]:
standard_error = np.sqrt(D/len(mothers) + D/len(daughters))
print(f'Стандартная ошибка - {standard_error: .4f}')

Стандартная ошибка -  3.8880


Расчитываем коэффициент Стьюдента: 

In [6]:
d_f=(len(mothers)-1) + (len(daughters) - 1)
print(f'Степени свободы для расчета коэффициента Стьдента -{d_f: .0f}')
t_critical = stats.t.ppf(0.975, d_f)
print(f'Коэффициент Стьюдента - {t_critical: .4f}')

Степени свободы для расчета коэффициента Стьдента - 18
Коэффициент Стьюдента -  2.1009


Расчитываем левую и правую границы интервала и получаем ответ:

In [7]:
l_border = difference_of_means - t_critical * standard_error
r_border = difference_of_means + t_critical * standard_error
print(f'Левая граница интервала  - {l_border: .2f}')
print(f'Правая граница интервала - {r_border: .2f}')
print(f'>>> 95% доверительный интервал для разности среднего роста родителей и детей: [{l_border:.2f}, {r_border:.2f}]')

Левая граница интервала  - -6.27
Правая граница интервала -  10.07
>>> 95% доверительный интервал для разности среднего роста родителей и детей: [-6.27, 10.07]


---  
Ответ:  
95% доверительный интервал для разности среднего роста родителей и детей: [-6.27, 10.07]

---